src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1054 @@
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->log_type = type;
+}
+
+void
+cub_ddl_log_execute_result (T_SRV_HANDLE * srv_handle)
+{
+  if (srv_handle == NULL)
+    {
+      return;
+    }
+
+  for (int i = 0; i < srv_handle->num_q_result; i++)
+    {
+      if (cub_is_ddl_type (srv_handle->q_result[i].stmt_type) == TRUE)
+	{
+	  cub_ddl_log_stmt_type (srv_handle->q_result[i].stmt_type);
+	  break;
+	}
+      else
+	{
+	  cub_ddl_log_stmt_type (srv_handle->q_result[i].stmt_type);
+	}
+    }
+}
+
+static int
+cub_file_copy (char *src_file, char *dest_file)
+{
+  char buf[FILE_BUFFER_SIZE] = { 0 };
+  size_t size;
+  int retval = 0;
+
+  if (src_file == NULL || dest_file == NULL)
+    {
+      return -1;
+    }
+
+  FILE *fsource = fopen (src_file, "r");
+  if (fsource == NULL)
+    {
+      return -1;
+    }
+
+  FILE *fdest = fopen (dest_file, "w");
+  if (fdest == NULL)
+    {
+      return -1;
+    }
+
+  while (retval == 0)
+    {
+      size = fread (buf, 1, FILE_BUFFER_SIZE, fsource);
+
+      if (size < 1 || size > FILE_BUFFER_SIZE)
+	{
+	  if (feof (fsource))
+	    {
+	      break;
+	    }
+	  else
+	    {
+	      retval = ferror (fsource);
+	    }
+	}
+      else
+	{
+	  if (fwrite (buf, 1, size, fdest) != size)
+	    {
+	      retval = ferror (fdest);
+	      break;
+	    }
+	}
+    }
+
+  if (fsource != NULL)
+    {
+      fclose (fsource);
+    }
+
+  if (fdest != NULL)
+    {
+      fclose (fdest);
+    }
+
+  return retval;
+}
+
+static int
+cub_make_schema_file_name (const char *file_full_path, char *dest_path, size_t buf_size)
+{
+  const char *env_root = NULL;
+  char time[TIME_STRING_SIZE] = { 0 };
+  const char *name_tmp = NULL;
+  char *tpath = NULL;
+  int retval = 0;
+
+  if (ddl_audit_handle == NULL || file_full_path == NULL || dest_path == NULL || buf_size < 0)
+    {
+      return -1;
+    }
+
+  env_root = envvar_root ();
+  cub_get_current_time (time, TIME_STRING_SIZE);
+
+  name_tmp = strrchr (file_full_path, PATH_SEPARATOR);
+
+#if defined(WINDOWS)
+  {
+    const char *nn_tmp = strrchr (file_full_path, PATH_SEPARATOR);
+    if (name_tmp < nn_tmp)
+      {
+	name_tmp = nn_tmp;
+      }
+  }
+#endif /* WINDOWS */
+  if (name_tmp == NULL)
+    {
+      name_tmp = file_full_path;
+    }
+  else
+    {
+      name_tmp++;
+    }
+
+  retval = snprintf (ddl_audit_handle->schema_file, buf_size, "%s_%s_%d", name_tmp, time, getpid ());
+  if (retval < 0)
+    {
+      assert (false);
+      ddl_audit_handle->schema_file[0] = '\0';
+      return retval;
+    }
+
+  retval =
+    snprintf (dest_path, buf_size, "%s/%s/%s_%s_%d", env_root, DDL_LOG_LOADDB_FILE_PATH, name_tmp, time, getpid ());
+
+  if (retval < 0)
+    {
+      assert (false);
+      dest_path[0] = '\0';
+      return retval;
+    }
+#if defined(WINDOWS)
+  unix_style_path (dest_path);
+#endif
+
+  tpath = strdup (dest_path);
+  if (tpath == NULL)
+    {
+      return -1;
+    }
+  cub_create_dir_log (tpath);
```

Comment:
I don't know why you duplicate the dest_path. And if cub_create_dir_log() returned a negative value, it seems you should use this. Remove from `tpath = strdup (dest_path);...`, and replace it:

`retval = cub_create_dir_log (dest_path);`